buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/trialsinterface.py

```diff
@@ -0,0 +1,74 @@
+from pathlib import Path
+
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils.json_schema import FolderPathType
+from pynwb.file import NWBFile
+
+from scipy.io import loadmat as loadmat_scipy
+
+
+def access_behavior_property_safe(property, parent, behavior_mat):
+    trial_info = behavior_mat["behavior"]["trials"]
+    nest_depth = len(trial_info["position_trcat"])
+
+    value = parent[property]
+    if nest_depth > 1:
+        value = [num for sublist in value for num in sublist]  # Flatten the list if large depth
+
+    assert len(value) == len(
+        trial_info["recordings"]
+    )  # Save access properties should have the same length as the number of recordings
+
+    return value
+
+
+# Add trial table from the behavior file
+class HuszarTrialsInterface(BaseDataInterface):
+    def __init__(self, folder_path: FolderPathType):
+        super().__init__(folder_path=folder_path)
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False):
+        self.session_path = Path(self.source_data["folder_path"])
+        self.session_id = self.session_path.stem
+
+        # Add trial table from the behavior file
+        behavior_file_path = self.session_path / f"{self.session_id}.Behavior.mat"
+        behavior_mat = loadmat_scipy(behavior_file_path, simplify_cells=True)
+
+        trial_info = behavior_mat["behavior"]["trials"]
+        trial_interval_list = access_behavior_property_safe("trial_ints", trial_info, behavior_mat)
+
+        data = []
+
+        for start_time, stop_time in trial_interval_list:
+            data.append(
+                dict(
+                    start_time=float(start_time),
+                    stop_time=float(stop_time),
+                )
+            )
+
+        [nwbfile.add_trial(**row) for row in sorted(data, key=lambda x: x["start_time"])]
+
+        nwbfile.add_trial_column(
+            name="choice",
+            description="An value of 0 or 1 representing whether the expected and visited arm of the trial matches",
```

Comment:
Really? So it might be more descriptive if this trial column was named `'visited_matched_expected'`

Also, for all these columns, are you saying that `0` is left and `1` is right? If so, I think we should choose one of two alternate representations so it's super clear and readable (not just just within descriptions)

a) replace the integers with strings 'left' and 'right' - being sure to compress the columns on the table, this should be fine from a data engineering POV

b) replace the integers with booleans and rename the columns to `'visited_left'` (or `'visited_right'` correspondingly); similarly `'choice'` -> `'matches_expected'`; `'expected_arm'` -> `'expected_right'`

Which do you prefer?

Reply:
I see. I was trying to keep the column names the same as the MATLAB field names but we don't have to really do that.

Hmm I'd go for (a). I'd decided to frame this as an integer value (used as a categorical label) because technically a maze could have more than two options—though, in this case, it doesn't and can be readily mapped to 'left' / 'right'. So I thought that would be the most encompassing approach, albeit slightly abstract.

Looking back, there is also technically a third option (NaN) that wouldn't fit into a strictly string or boolean typing system unless you think NaN or None is alright to have here. How would you suggest handling that?

I also don't know whether 0 means left or right. Should that be something confirmed with the author?

Reply:
> Looking back, there is also technically a third option (NaN) that wouldn't fit into a strictly string or boolean typing system unless you think NaN or None is alright to have here. How would you suggest handling that?

`NaN` in that context would mean "neither" or "timeout" condition

> I also don't know whether 0 means left or right. Should that be something confirmed with the author?

You can maybe check the spatial series through the widgets aligned to around the time of such a trial event to try to determine that